pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -83,21 +83,22 @@ def _check_consider_iterating_dictionary(self, node: nodes.Call) -> None:
             return
         if node.func.attrname != "keys":
             return
+        comp_ancestor = utils.get_node_first_ancestor_of_type(node, nodes.Compare)
         if (
             isinstance(node.parent, (nodes.For, nodes.Comprehension))
-            or isinstance(node.parent, nodes.Compare)
+            or isinstance(comp_ancestor, nodes.Compare)
```

Comment:
```suggestion
            or comp_ancestor
```

The `isinstance` is unnecessary now that we use `utils.get_node_first_ancestor_of_type`.